scipy/sparse/base.py

```diff
@@ -140,6 +140,19 @@ def getmaxprint(self):
     #        typ = None
     #    return typ
 
+    def count_nonzero(self):
+        """Number of non-zero entries, equivalent to
+
+        np.count_nonzero(a.toarray())
+
+        Unlike getnnz and the nnz property, which return the number of stored
+        entries (the length of the data attribute), this method counts the
+        actual number of non-zero entries in data.
+        """
+        # Subclasses may implement more efficient versions of this method,
+        # but this acts as a generic fallback in case they do not.
```

Comment:
Do we have subclasses that don't implement this?
If not, maybe prefer to raise NotImplementedError?